lib/vehicle.rb

```diff
@@ -22,4 +26,19 @@ def antique?
   def electric_vehicle?
     @engine == :ev
   end
+
+  def set_registration_date
+    @registration_date = Date.today
+  end
+  
+  def set_plate_type
+    if antique? == true
+      @plate_type = :antique
+    elsif electric_vehicle? == true
+      @plate_type = :ev
+    else
+      @plate_type = :regular
+    end
+  end
```

Comment:
I love these two methods you created here in the Vehicle class. Since we're changing a vehicle attribute, it makes sense that the class that holds those attributes should be in charge of changing those attributes.